### DISCUSSION TITLE
Implementing a URL handler for web based initiation of install

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [Devolutions/UniGetUI#2654] [General]
## Background

I was thinking about how I can share a list of my application in an eye pleasing way with my peers (or future self who is installing a new system).
Scripting is one way, but every system is a bit different, and so picking and choosing is probably the correct way.

There are bundles, but that requires a buy-in just to view the list properly, and so, a web page full of links seem like a good way to start a conversation with someone about what I have installed on my system (yes, sorry, I'm that type of nerd).

## Possible Scenario # 1

So I thougth to myself: "It would be great if a link would trigger UniGetUI, present the (already amazing) package detail with that beautiful INSTALL button at the bottom" - basically removing the friction from guiding someone through an installation process, allowing full control.

The all I need to do is share a simple HTML file with a friend, and have them install whatever they want following the list from their browser.

## Possible Scenario # 2

This is a bit more far-fetched, because it's not only about UniGetUI.

If recently stumbled upon [winget.run](https://winget.run), and searching the repo, I see a few issues/discussions where marticliment, the mainainer, acknowledges their work.

I was thinking these projects, since they server similar audiences, should somehow be loosly coupled in a way that they will augment one another.

Having a link one can click on `winget.run` and it opens up UniGetUI's package detail would be awesome.

The benfit `winget.run` can get from cooperation is to utilize the user generated metadata UniGetUI has already started gathering through the [Icons and Screenshots](https://www.marticliment.com/unigetui/help/#docs) database.

## The propsal

* Phase # 1: Implement a URL handler for UniGetUI that will allow installing packages from any of the supported source.
* Phase # 2: Allow the URL handler to pass installation arguments to control the install process (local user only, force interactive, etc.)
* Phase # 3: Open bundles from online sources directly from a webpage.

I'm not a talented coder, but I'm willing to try my chops here - with some guidance, unless this is something that will interest marti and he decides to take this as a challenge.

This is all very 'raw' thought, and I might be missing some obvious problems.
I'd like to get some feedback on whether this has the potential I think it has?
Would this be something that would properl the project further or be a dead weight?

**Comment by marticliment** [comment]
I will convert this to a feature request.

My idea is to have a custom url (`unigetui:`) that launches UniGetUI and can trigger certain actions.
